docs/developers/components/managing.md

```diff
@@ -0,0 +1,97 @@
+
+### Adding components to root
+
+Similar to how components can specify other components within their **config.yaml**, components can be added to Harper by adding them to the **harperdb-config.yaml**.
+
+The configuration is very similar to that of **config.yaml**. Entries are comprised of a top-level `<name>:`, and an indented `package: <specifier>` field. Any additional component option can also be included as indented fields.
+
+```yaml
+status-check:
+  package: "@harperdb/status-check"
+```
+
+The key difference between this and a component's **config.yaml** is that the name does **not** need to be associated with a **package.json** dependency. When Harper starts up, it transforms these configurations into a **package.json** file, and then executes a form of `npm install`. Thus, the `package: <specifier>` can be any valid dependency syntax such as npm packages, GitHub repos, tarballs, and local directories are all supported.
+
+Given a root config like:
+
+```yaml
+myGithubComponent:
+  package: HarperDB-Add-Ons/package#v2.2.0 # install from GitHub 
+myNPMComponent:
+  package: harperdb # install from npm
+myTarBall:
+  package: /Users/harper/cool-component.tar # install from tarball
+myLocal:
+  package: /Users/harper/local # install from local path
+myWebsite:
+  package: https://harperdb-component # install from URL
+```
+
+Harper will generate a **package.json** like:
+
+```json
+{
+  "dependencies": {
+    "myGithubComponent": "github:HarperDB-Add-Ons/package#v2.2.0",
+    "myNPMComponent": "npm:harperdb",
+    "myTarBall": "file:/Users/harper/cool-component.tar",
+    "myLocal": "file:/Users/harper/local",
+    "myWebsite": "https://harperdb-component"
+  }
+}
+```
+
+npm will install all the components and store them in `<rootPath>/<componentsRoot>`. A symlink back to `<rootPath>/node_modules` is also created for dependency resolution purposes.
+
+The package prefix is automatically added, however you can manually set it in your package reference.
+
+```yaml
+myCoolComponent:
+  package: file:/Users/harper/cool-component.tar
+```
+
+By specifying a file path, Harper _should_ generate a symlink from that location to the **componentsRoot**. In case the symlink is not being generated, the final, advanced method for component loading is to manually create symlinks from the local component source directory to the Harper instance **componentsRoot**. For most UNIX machines this is as simple as: `ln -s <path/to/component/source> <componentsRoot>`.
+
+For example, if the machine had a **componentsRoot** of `/hdb/components`, and the component source was located at `/Users/dev/my-custom-component`, the symlink command would be: `ln -s /Users/dev/my-custom-component /hdb/components`
```

Comment:
Inline comment instead of bold?

```suggestion
For example, if the machine had a `componentsRoot` of `/hdb/components`, and the component source was located at `/Users/dev/my-custom-component`, the symlink command would be: `ln -s /Users/dev/my-custom-component /hdb/components`
```